modin/core/execution/ray/generic/io/io.py

```diff
@@ -134,56 +148,51 @@ def func(df, **kw):
                 `partition_idx` serving as chunk index to maintain rows order.
             """
-            if kw["partition_idx"] != 0:
+            idx = kw["partition_idx"]
+            _kwargs = kwargs.copy()
+            if idx != 0:
                 # we need to create a new file only for first recording
                 # all the rest should be recorded in appending mode
-                if "w" in kwargs["mode"]:
-                    kwargs["mode"] = kwargs["mode"].replace("w", "a")
+                if "w" in _kwargs["mode"]:
+                    _kwargs["mode"] = _kwargs["mode"].replace("w", "a")
                 # It is enough to write the header for the first partition
-                kwargs["header"] = False
+                _kwargs["header"] = False
 
             # for parallelization purposes, each partition is written to an intermediate buffer
-            path_or_buf = kwargs["path_or_buf"]
-            is_binary = "b" in kwargs["mode"]
+            path_or_buf = _kwargs["path_or_buf"]
+            is_binary = "b" in _kwargs["mode"]
             if is_binary:
-                kwargs["path_or_buf"] = io.BytesIO()
+                _kwargs["path_or_buf"] = io.BytesIO()
             else:
-                kwargs["path_or_buf"] = io.StringIO()
-            df.to_csv(**kwargs)
-            content = kwargs["path_or_buf"].getvalue()
-            kwargs["path_or_buf"].close()
-
-            # each process waits for its turn to write to a file;
-            # in case of violation of the order of receiving messages from the queue,
-            # the message is placed back
-            while True:
-                get_value = queue.get(block=True)
-                if get_value == kw["partition_idx"]:
-                    break
-                queue.put(get_value)
+                _kwargs["path_or_buf"] = io.StringIO()
+            df.to_csv(**_kwargs)
+            content = _kwargs["path_or_buf"].getvalue()
+            _kwargs["path_or_buf"].close()
+
+            # each process waits for its turn to write to a file
+            ray.get(signals.wait.remote(idx))
 
             # preparing to write data from the buffer to a file
             with pandas.io.common.get_handle(
                 path_or_buf,
                 # in case when using URL in implicit text mode
                 # pandas try to open `path_or_buf` in binary mode
-                kwargs["mode"] if is_binary else kwargs["mode"] + "t",
-                encoding=kwargs["encoding"],
-                errors=kwargs["errors"],
-                compression=kwargs["compression"],
-                storage_options=kwargs["storage_options"],
+                _kwargs["mode"] if is_binary else _kwargs["mode"] + "t",
+                encoding=_kwargs["encoding"],
+                errors=_kwargs["errors"],
+                compression=_kwargs["compression"],
+                storage_options=_kwargs["storage_options"],
```

Comment:
you haven't changed any of those used parameters in `_kwargs`, why don't you just use original `kwargs`?..